### DISCUSSION TITLE
Understanding prometheus metrics

### DISCUSSION THREAD

**Discussion by obh** [prisma/prisma#15894] [Q&A] [answered]
I have added prometheus metrics to my application which uses prisma. However, I see quite a lot latency in some metrics published by prisma. Here is a snapshot of the histogram

```
# HELP prisma_client_queries_duration_histogram_ms Histogram of the duration of all executed Prisma Client queries in ms
# TYPE prisma_client_queries_duration_histogram_ms histogram
prisma_client_queries_duration_histogram_ms_bucket{le="0"} 0
prisma_client_queries_duration_histogram_ms_bucket{le="1"} 89
prisma_client_queries_duration_histogram_ms_bucket{le="5"} 115
prisma_client_queries_duration_histogram_ms_bucket{le="10"} 12049
prisma_client_queries_duration_histogram_ms_bucket{le="50"} 13562
prisma_client_queries_duration_histogram_ms_bucket{le="100"} 13568
prisma_client_queries_duration_histogram_ms_bucket{le="500"} 13568
prisma_client_queries_duration_histogram_ms_bucket{le="1000"} 13568
prisma_client_queries_duration_histogram_ms_bucket{le="5000"} 13568
prisma_client_queries_duration_histogram_ms_bucket{le="50000"} 13568
prisma_client_queries_duration_histogram_ms_bucket{le="+Inf"} 13568
prisma_client_queries_duration_histogram_ms_sum 113660.23126399997
prisma_client_queries_duration_histogram_ms_count 13568
```

as you can see the latency here seems to be quite high. This application is just doing an insert into a mysql table with no indexes. I checked the database metrics and found that the real inserts are quite fast, < 1ms. I also logged the duration of the queries by hooking into `primsa.$on("query"...bit). The duration that I get in these logs is also very low, < 2ms. Here is a snapshot. 
```
prisma.$on("query", async (e) => {
    if(Date.now() % 17 == 0){
        // logger.log('Query: ' + e.query)
        logger.log('Duration: ' + e.duration + 'ms')
    }
})
```

<img width="833" alt="image" src="https://user-images.githubusercontent.com/677851/196784839-d5623307-4845-4f8b-a2dc-435c182eaffa.png">

Can someone please help me understand why this diff?

**Comment by garrensmith** [comment]
@obh could you give me a bit more context. What does your schema look like and what does the code look like that is executing the queries? This could be a bug but I need a bit more information to confirm.

**Comment by obh** [comment, marked as answer]
@garrensmith Figured out the issue at my end (now everything is as expected). My mistake was in assuming that the 'query' on handler returns a single response, but for a single create request the series of response is - BEGIN, INSERT, SELECT, COMMIT. 

Overall, just to share some numbers, I am seeing around `8ms` as the insert time for a table with 8 columns and only a primary key (mysql).  Mostly running a throughput of around 20-30 inserts per second.

**Comment by garrensmith** [comment]
Awesome @obh that is good to hear. Let me know if you have any other questions.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
